addon/mixins/routes/application.js

```diff
@@ -31,6 +31,13 @@ export default Ember.Mixin.create({
         outlet: outlet,
         parentView: parentView
       });
+
+      // Check if another modal exists then give it focus
+      Ember.run.later(function() {
```

Comment:
I didn't try the other options to run on the run loop. I'm not too familiar yet but when running outside of it, the `$` was returning also the element that was going to be removed.